### PR TITLE
Updates for user guide and reference guide

### PR DESCRIPTION
**Pull request by aireilly**
Ready for review/merge. Local build LGTM.

Changes:

* Numeruous content tweaks, capitalization, opinionated text changes
* Added a Red Hat nav bar logo
* Added a GitHub Icon to source code link
* Refactored away partials in user guide - they are convoluted and not required. 
* New file `./tools/creat_ssg_refs.py`. First pass at generated Error and Suggestion tables. Final results were tidied up by hand, still WIP, but output is acceptable.

### CODE REVIEW

File: modules/end-user-guide/nav.adoc [hunk 1]
```diff
@@ -1,10 +1,10 @@
-.End-user guide
+.User guide
```
Comment (themr0c) [inline modules/end-user-guide/nav.adoc:1]: Ideally, we should also rename the module and add redirections. We can do it in a subsequent pull request.

Comment (aireilly) [reply modules/end-user-guide/nav.adoc:1]: hmm. I can do this now, may as well.

Comment (aireilly) [reply modules/end-user-guide/nav.adoc:1]: done

**Review by rolfedh — APPROVED** [review]

**Comment by rolfedh** [comment]
I'm also grateful to you @aireilly for modeling the best practices we should follow: Describing the changes and getting reviews from the team.

File: modules/end-user-guide/pages/adding-vale-configuration-to-a-project.adoc [hunk 2]
```diff
@@ -1,18 +1,49 @@
 // Metadata for Antora
-:navtitle: Adding to a project
+:navtitle: Adding vale to a project
```
[the remark this answers is not in the document]
Comment (aireilly) [reply modules/end-user-guide/pages/adding-vale-configuration-to-a-project.adoc]: yes - I applied this style change throughout.

**Pull request merged**